### PR TITLE
Ncs 656 validation status endpoint

### PR DESCRIPTION
**Pull request by mattch1**
Checks each field to ensure that they are present and that they have the correct status value.
Asked analyst and we will be including recent filing.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/controller/ConfirmationStatementController.java [hunk 1]
```diff
@@ -58,4 +60,15 @@ public ConfirmationStatementController(ConfirmationStatementService confirmation
         return serviceResponse.<ResponseEntity<Object>>map(ResponseEntity::ok)
                 .orElseGet(() -> ResponseEntity.notFound().build());
     }
+
+    @GetMapping("/{confirmation_statement_id}/validation-status")
+    public ResponseEntity<Object> getValidationStatus(@PathVariable("confirmation_statement_id") String submissionId) {
+        try {
+            ValidationStatusResponse validationStatusResponse = confirmationStatementService.areTasksComplete(submissionId);
+            return ResponseEntity.ok().body(validationStatusResponse);
+        } catch (CompanyNotFoundException e) {
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/confirmationstatementapi/controller/ConfirmationStatementController.java:69]: i'd change the exception type to Exception just to make sure we catch any other exceptions apart from CompanyNotFoundException. You could add it after this block but looks like they'll both be printing the same message so probably better just to change the existing catch block.

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/confirmationstatementapi/controller/ConfirmationStatementController.java:69]: actually, stick the extra catch(Exception e) after this block and have a generic message. The message in CompanyNotFoundException is relating to submission not found, perhaps change this message to say Could not find company.

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementService.java [hunk 2]
```diff
@@ -132,6 +137,45 @@ private void makePayableResourceIfUnpaid(String createdUri,
         }
     }
 
+    public ValidationStatusResponse areTasksComplete(String submissionId) throws CompanyNotFoundException {
+        Optional<ConfirmationStatementSubmissionJson> submissionJsonOptional = getConfirmationStatement(submissionId);
+        if (submissionJsonOptional.isEmpty()) {
+            throw new CompanyNotFoundException();
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementService.java:143]: CompanyNotFoundException is probably not the correct one to throw here. Maybe use a ServiceException with a message indicating why it failed. Or if you want a specific one, create a SubmissionNotFound one?

File: src/test/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementServiceTest.java [hunk 3]
```diff
@@ -246,6 +248,59 @@ void getConfirmationSubmissionNotFound() {
         assertFalse(result.isPresent());
     }
 
+    @Test
```
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/confirmationstatementapi/service/ConfirmationStatementServiceTest.java:251]: i'd also add a test in here for when submissionData is null (to cover line 151 in confirmationStatementService)

**Review by markpit — APPROVED** [review]
